cmd/repo.go

```diff
@@ -283,9 +267,74 @@ func readOrUpdateIndex() repo.Index {
 		art.SetUninstaller(verbose, ercoleConfig.RepoService.DistributedFiles)
 	}
 
+	index = append(index, getArtifactsNotIndexed(index, ercoleConfig.RepoService.DistributedFiles)...)
+
+	index.SortArtifactInfo()
+
 	return index
 }
 
+// getArtifactsNotIndexed scan filesystem for installed artifacts not in index
+func getArtifactsNotIndexed(index repo.Index, distributedFiles string) repo.Index {
+	filesNotIndexed := getFilesNotIndexed(index, distributedFiles)
+
+	artifactsNotIndexed := make(repo.Index, 0)
+	for _, file := range filesNotIndexed {
+		artifactInfo := new(repo.ArtifactInfo)
+		artifactInfo.Filename = filepath.Base(file.Name())
+
+		if err := artifactInfo.SetInfoFromFileName(artifactInfo.Filename); err != nil {
+			continue
+		}
+
+		artifactInfo.Repository = "???"
+		artifactInfo.Installed = true
+		artifactInfo.ReleaseDate = file.ModTime().Format("2006-01-02")
+		artifactInfo.UpstreamType = "???"
+
+		artifactsNotIndexed = append(artifactsNotIndexed, artifactInfo)
+	}
+
+	return artifactsNotIndexed
+}
+
+func getFilesNotIndexed(index repo.Index, distributedFiles string) []os.FileInfo {
+	installedInIndex := make(map[string]bool)
+
+	for _, artifact := range index {
+		if artifact.Installed {
+			installedInIndex[artifact.FilePath(distributedFiles)] = true
+			installedInIndex[filepath.Join(distributedFiles, "all", artifact.Filename)] = true
+		}
+	}
+
+	filesNotIndexed := make([]os.FileInfo, 0)
+	err := filepath.Walk(distributedFiles,
+		func(filePath string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if installedInIndex[filePath] {
+				return nil
+			}
+
+			fileInfo, err := os.Stat(filePath)
+			if err != nil || fileInfo.IsDir() {
+				return nil
+			}
+
+			filesNotIndexed = append(filesNotIndexed, fileInfo)
+
+			return nil
+		})
```

Comment:
You are indexing too much files! You need to index only artifacts in /all.

Reply:
Some files should not be indexed (e.g index.json, shared/*) and I suspect you have duplicates in the index

Reply:
Why index only the files in `/all`? If a file is inserted elsewhere isn't important?
This method do not insert this files in the index, in fact index.json etc aren't indexed.
This list is passed to getArtifactsNotIndexed that checks each file if is an artifact, so if they match against any {ercole-agent,ercole} regex.
There aren't files duplicated here, but there are duplicates from `/all` and specific folders yes, but they are both removed later.

Reply:
Because I may want to have multiple ercole-agent distribution, which the filename is the same but that the content may differ a little e.g. with different ercole-agent position...

Reply:
Because other files should not be managed by ercole or because it's a link (or metadata) to /all.

Reply:
I don't want to rely on artifacts regexes...

Reply:
Okay, I update and:
I do not rely on regexes.
I check only `/all` folder.